pkg/image/api/types.go

```diff
@@ -60,6 +62,93 @@ type ImageLayer struct {
 	Size int64
 }
 
+const (
+	// The supported type of image signature.
+	ImageSignatureTypeAtomic string = "atomic"
+)
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// Conditions represent the latest available observations of a signature's current state.
+	Conditions []SignatureCondition
+
+	// Following metadata fields will be set by server if the signature content is successfully parsed and
+	// the information available.
+
+	// A human readable string representing image's identity. It could contain company's name, a product name
+	// and a version. Eligible value is image's pull specification
```

Comment:
I think "a product name and version, or a image pull spec." reads a little better.